src/frontend/src/expr/correlated_input_ref.rs

```diff
@@ -26,9 +26,9 @@ use super::Expr;
 ///
 /// `index` is the index in the refered relation.
 pub struct CorrelatedInputRef {
-    index: usize,
-    data_type: DataType,
-    depth: usize,
+    pub index: usize,
+    pub data_type: DataType,
+    pub depth: usize,
```

Comment:
Seems only needed by `get_and_change_correlated_input_ref` to set index. Given it is a relatively lightweight leaf node, what about `*correlated_input_ref = CorrelatedInputRef::new(...)` without exposing the fields?